pkg/ebpf/c/tracee.bpf.c

```diff
@@ -656,6 +656,306 @@ int tracepoint__sched__sched_process_fork(struct bpf_raw_tracepoint_args *ctx)
+    u64 empty = 0;
+    bpf_map_update_elem(&already_reported_hidden_modules_map, &mod_addr, &empty, BPF_ANY);
+}
+
+// Populate the queue with the current modules in the list of modules.
+// We can't read it once and then hook on do_init_module and free_module since a hidden module will
+// remove itself from the list directly and we wouldn't know (hence from our perspective the module
+// will reside in the modules list, which could be false). So on every trigger, we go over the
+// modules list and populate the queue. This is a queue since we need to delete all elements so the
+// next run will be clean. Since this mechanism is suppose to be triggered every once in a while,
+// this should be ok.
+static __always_inline bool init_shown_modules_queue()
+{
+    struct list_head *head = (struct list_head *) get_symbol_addr("modules");
+    bool iterated_all_modules = false;
+    struct module *pos, *n;
+
+    pos = list_first_entry(head, typeof(*pos), list);
+    n = list_next_entry(pos, list);
+
+#pragma unroll
+    for (int i = 0; i < MAX_NUM_MODS; i++) {
+        if (&pos->list == head) {
+            iterated_all_modules = true;
+            break;
+        }
+
+        u64 empty = 0;
+        bpf_map_update_elem(&unhidden_modules_map, &pos, &empty, BPF_ANY);
+
+        pos = n;
+        n = list_next_entry(n, list);
+    }
+
+    return !iterated_all_modules; // false is valid value
+}
+
+static __always_inline bool is_hidden(u64 mod)
+{
+    return bpf_map_lookup_elem(&unhidden_modules_map, &mod) ==
+           NULL; // if we don't find the module in our map, it means the module is hidden
+}
+
+static __always_inline bool find_modules_from_module_kset_list()
+{
+    struct kset *mod_kset = (struct kset *) get_symbol_addr("module_kset");
+    struct list_head *head = &(mod_kset->list);
+    struct kobject *pos = list_first_entry(head, typeof(*pos), entry);
+    struct kobject *n = list_next_entry(pos, entry);
+    bool finished_iterating = false;
+
+#pragma unroll
+    for (int i = 0; i < MAX_NUM_MODS; i++) {
+        if (&pos->entry == head) {
+            finished_iterating = true;
+            break;
+        }
+
+        if (READ_KERN(n->name) == NULL) { // Without this the list seems infinite. Also, using pos
+                                          // here seems incorrect as it starts from a weird member
+            finished_iterating = true;
+            break;
+        }
+
+        struct module_kobject *mod_kobj =
+            (struct module_kobject *) container_of(n, struct module_kobject, kobj);
+        if (mod_kobj) {
+            struct module *mod = READ_KERN(mod_kobj->mod);
+            if (mod) {
+                if (is_hidden((u64) mod)) {
+                    update_hidden_modules((u64) mod, mod->name);
+                }
+            }
+        }
+
+        pos = n;
+        n = list_next_entry(n, entry);
+    }
+
+    return !finished_iterating;
+}
+
+static __always_inline u32 prepare_hidden_mods_output(u64 *addrs,
```

Comment:
by sending an event per each discovered hidden module, this (and other) step can be removed